javalib/src/main/scala/java/net/URI.scala

```diff
@@ -839,8 +853,8 @@ object URI {
         if (diff != 0) diff
         else if (x.charAt(i) == '%') {
           // we need to do a CI compare for the next two characters
-          assert(x.length > i + 2, "Invalid escape in URI")
-          assert(y.length > i + 2, "Invalid escape in URI")
+          if (i + 3 > x.length || i + 3 > y.length)
```

Comment:
Maybe use `i + 2 >= x.length` for better readability (the 2 has the semantic value of the escape length). Also below.